### PR TITLE
Add support for release handling

### LINKED ISSUE #6: RFE: Support for the release option for subscription manager
As part of standing up a specific version of our solution, I need to run the following BEFORE adding the repos we want to subscribe to;
_**subscription-manager release --set=7.4**_

Being able to set this option in the role would be extremely helpful

### PR DESCRIPTION
**Pull request by tehsmyers**
Specifically, this adds support for the `--set` and `--unset` flags to
the `subscription-manager release` subcommand.

Closes #6

### CODE REVIEW

**Comment by tehsmyers** [comment]
I'd like to test the hell out of this before merging.

**Comment by tehsmyers** [comment]
I don't know why travis is failing, looks like an incompatibility between how molecule (or the underlying ansible module) is trying to call into docker versus the docker api. The downstream tests have not been updated to cover this, but their passing at least means that this change doesn't break other tasks. :)

File: README.md [hunk 1]
```diff
@@ -92,11 +97,11 @@ Role Output
 
 The `oasis_role_rhsm` fact will be set by this role, containing the following outputs:
 
-- `repositories` - The complete list of subscribable repositories known to the system, as well as their enabled status.
-  This value will be `false` if the `rhsm_repositories` input var is not used.
 - `subscribed` - Whether or not the system is registered. (bool)
-- `subscribed_pool_ids` - A list of pool IDs current attached too the system. Will be an empty list if no pools or attached,
+- `subscribed_pool_ids` - A list of pool IDs current attached too the system. Will be an empty list if no pools are attached,
```
Comment (robled) [inline README.md]: typo - 'too'

File: defaults/main.yml [hunk 2]
```diff
@@ -68,6 +68,13 @@ rhsm_force_register: "{{ omit }}"
 # yes/no bool, default no
 rhsm_unregister: false
 
+# Bool, whether ot not to unset the subscription-manager release,
```
Comment (robled) [inline defaults/main.yml]: typo - 'ot'

**Review by robled — DISMISSED** [review]

File: tasks/release.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# TODO These tasks should be considered for addition to the ansible
+#      rhsm_repositories module, which would remove the need to set facts to
+#      ensure idempotence when using the command module.
+- name: Unset subscription-manager release
+  command: subscription-manager release --unset
+  when: rhsm_release_unset and oasis_roles_rhsm_release
+  register: rhsm_release_unset_task
```
Comment (greg-hellings) [inline tasks/release.yml:7]: On this task and the next, is there any way to determine if changes were made? Sometimes there's a way to set `changed_when` based on the stdout of the command or such?

File: tasks/release_fact.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+- name: Gather subscription-manager release information
+  command: subscription-manager release --show
+  register: rhsm_release_check
+  changed_when: false
+  failed_when: false
```
Comment (greg-hellings) [inline tasks/release_fact.yml:5]: If `failed_when: false` should this also have `ignore_errors: true`? Or do we still want to capture non-zero exit statuses from the command?

File: tests/yamllint.yml [hunk 5]
```diff
@@ -3,6 +3,7 @@ extends: default
 rules:
   line-length:
     level: error
+    allow-non-breakable-inline-mappings: true
```
Comment (greg-hellings) [inline tests/yamllint.yml]: What does this do, and is it something we should include in the meta_skeleton for future roles?

Comment (tehsmyers) [reply tests/yamllint.yml]: It allows line-length to exceed the default maximum in non-breakable strings in inline mappings. In my case, it lets me put longish urls into a playbook without having to mangle them or do other trickery. This is for a specific downstream test scenario, though, so it's probably worth using only in that scenario. I do not think it's worth including in the meta_skeleton by default, but I do think it's good to know about.

tl;dr I'm going to remove this. :)

File: tasks/release_fact.yml (hunk 4 above)
Comment (tehsmyers) [reply tasks/release_fact.yml:5]: We do still want to capture non-zero exist statuses from the command, it's used on line 21.

File: tasks/release.yml (hunk 3 above)
Comment (tehsmyers) [reply tasks/release.yml:7]: Sadly, no. Here's the output I'm working with:
```
# subscription-manager release --show
Release: 7.0
# subscription-manager release --set 7.0
Release set to: 7.0
# subscription-manager release --unset
Release preference has been unset
# subscription-manager release --unset
Release preference has been unset
```

So you can see we still get the "Release set to" text, even when the release was already set to the "new" value. The same is true when using `unset`; s-m reports that a change was made, even when that change is to the current value.

Comment (robled) [reply tasks/release.yml:7]: I suggest that we embellish the comment a bit to explain the unfortunate behavior here.  Sounds like getting this into the module itself is the real path forward.  Until then we can have some info to explain why we can't track actual changes here.

Comment (tehsmyers) [reply tasks/release.yml:7]: I'm not sure what you're asking for. The unfortunate behavior is that we have to set facts internally to ensure idempotence, and the fix is to put the behavior into the rhsm_repository module. That's what the comment already says.

I don't think it's worthwhile to write a 20-line comment in the role to explain this; I was planning to open up a github issue spelling out the problem fully once I actually had code in this repo to link to.

Comment (robled) [reply tasks/release.yml:7]: I was asking for info in the code comments about the subscription manager output, similar to your earlier comment in this thread.  As someone who hasn't done much with s-m, this seemed useful to me.  However, a GitHub issue with that info would also suffice.

Comment (tehsmyers) [reply tasks/release.yml:7]: Cool. I'll make sure to reference this comment in the issue and see about providing a more complete picture of how s-m is working for us (and against us) here, and how we can get this taken care of upstream.
